### ISSUE TITLE
RetopoFlow 2.0 is unusably slow

### ISSUE THREAD

**Issue by Danial79** [CGCookie/retopoflow#448]
Hi,

I just opened up the 2.0 Beta for the first time and tried it on a sculpt I had at hand. The sculpt is only 850k faces (1.7m tris) and I'm getting a warning in the tool panel saying "Target polycount is high! RetopoFlow may load slowly." It certainly was slow—to the point of being unusable. Not only did it take forever to start up, but my mesh shows up as a white wireframe which I can't even see the shape of (screenshot attached), and I keep getting beachballs just simply moving the mouse.

Computer specs:
iMac 27-inch
macOS High Sierra 10.13.4
3.2GHz Intel Core i5
16GB 1600MHz DDR3
NVIDIA GeForce GT 755M 1GB

![screen shot 2018-04-01 at 9 07 22 pm](https://user-images.githubusercontent.com/32725142/38172546-61a0b834-35f1-11e8-822b-c354e15e5984.png)

**Comment by vxlcoder** [comment]
Thank you for the report, @Danial79.

I believe you had your sculpted mesh selected when you started RF. This is your problem.

The selected mesh is considered as the target mesh (the new, low poly count retopo'd mesh).  Since you're starting a new retopo session, make sure to not have any mesh selected.

Read through the quick start guide for me details, but also let us know if you continue to have problems.

Thanks!

**Comment by Danial79** [comment]
Ah, sorry. I did read the quick start, but must have missed the part about not selecting anything. Anyway, that has solved the white mesh issue, though I'm still getting a message saying "Source polycount is high", and I'm getting a refresh rate of about 2-6 seconds, which you obviously can't do anything with. I did manage to make a single polystrip, so functionally it's working, it's just the crazy slow-down that's the issue. I don't have any slow-down at all in Blender or Substance Painter—even with meshes more complex than this one—so I don't think it's a hardware issue.

**Comment by vxlcoder** [comment]
Do you have a Retina display? We have getting some reports about RF running slowly on Retina displays.

**Comment by vxlcoder** [comment]
Hmmm.

I have access to a Mac, but not until tomorrow. I'll take a closer look at this issue.

If possible, could you attach your .blend file so that I can try to replicate your issue as closely as possible? There might be a Blender setting that is causing the slow down.

**Comment by Danial79** [comment]
I tried uploading it, but it's a 40MB ZIP, so hopefully [this link](https://www.dropbox.com/s/mwbkjt1nl4e7h1y/Sand-Dweller_WIP_Retopo.blend.zip?dl=0) to my Dropbox works. As an extra point of reference, I tried RF on a new scene with a default sphere, and there was still a tiny bit of lag, but it _was_ usable.

**Comment by carter2422** [comment]
@Danial79 @QuietRezo I think I found the source of the issue. I believe it's due to Blender's Window Draw Method.

Please try changing the draw method to `Overlap` via *File > User Preferences > System > Window Draw Method*.

Then try RetopoFlow again. 

Another way to test this is to load a default scene with the cube and press ALT+A for animation playback. If the draw method is correct for your machine then you should reach 24-60 FPS no problem (depending on the animation FPS chosen in in *Render > Frame Rate*).

**Comment by Danial79** [comment]
I just switched from Automatic to Overlap and Blender started going crazy. While in RF, I was getting a refresh of about 1 second, which is better than the 4-ish seconds I was getting before (still unusable), but the Properties Editor was also flashing between unshaded every refresh also. On top of that, when I exited RF, it persisted in the 3D view whenever my mouse was over it (if I moved the mouse to a side panel, it disappeared). Needless to say, I've had to switch back to Automatic. With regards to animation, I'm getting 60 FPS with the default cube while on Automatic Window Draw.

**Comment by carter2422** [comment]
@Danial79 I experienced that UI flashing while testing as well, but it was limited to certain Draw Methods.

What worked for me was loading Factory Settings, wiping out any conflicting configurations, and then setting the draw method to Automatic and starting RetopoFlow. 

Regardless, I'm going to do some testing with the model you submitted.

One last thing, what's the RetopoFlow FPS on the default cube?

**Comment by carter2422** [comment]
One more thing to try @Danial79. Does RetopoFlow speed up dramatically on the second start?

In other words:
1. Open your Sand Dweller, deselect everything
2. Start RetopoFlow; should be slow and take a little while
3. Exit RetopoFlow
4. Restart RetopoFlow; it should now be much faster as the source meshes are now cached

Can you confirm this is the case for you?

**Comment by Danial79** [comment]
When I tested it on the default sphere, I was manually counting a refresh about every second. Is there a way to see the actual FPS though? Haven't been at my computer for a few days so haven't been able to test if it's faster upon second loading, sorry. I'll get back to you on that one.

**Comment by vxlcoder** [comment]
@Danial79, the FPS shows under the Advanced section in the Options window.

![image](https://user-images.githubusercontent.com/3790699/38457305-7471986a-3a5c-11e8-9851-01187d2d7afd.png)

**Comment by vxlcoder** [comment]
Also, @Danial79, I just pushed a small commit to the `release/2.0.0` branch.  If you know how to use git, would you try the latest commit to see if it improves your performance?  To use, start RF, uncheck the `Async Loading` option under `Advanced` options section, then restart Blender.  You may notice that starting RF will be delayed with large meshes (source and/or target), but my hope is that this _may_ increase overall FPS once RF fully starts.

(Unfortunately, since I cannot reproduce it on any of my machines, I'm testing blind here... So, this might not fix the issue)

![image](https://user-images.githubusercontent.com/3790699/38458038-c3e36d10-3a66-11e8-9407-01a8f10ab25d.png)

**Comment by QuietRezo** [comment]
I'm trying out the new commit, but restarting RF with Async Loading disabled didn't improve much on the High Sierra Mid-2012 Retina MBP. 

<img width="1920" alt="fpsretopo" src="https://user-images.githubusercontent.com/38005618/38462080-c80aa4a0-3aa5-11e8-846e-5a57645c0a1b.png">

Here are my results on the default cube (It goes from 25 to 40 (I'm pretty sure I once saw 60) fps, even after restarting RF on the same default cube.

<img width="709" alt="screen shot 2018-04-07 at 8 50 38 pm" src="https://user-images.githubusercontent.com/38005618/38462085-d7f1a18e-3aa5-11e8-8f5f-e4b631ae0bab.png">
<img width="728" alt="screen shot 2018-04-07 at 8 53 06 pm" src="https://user-images.githubusercontent.com/38005618/38462086-d805bf3e-3aa5-11e8-8adf-90be37683078.png">

Using RF 2.0.0 in Windows PC's has been great, but I can't use it so far in the OSX laptop while moving around. I'll try it out in my brother's desktop OSX next weekend, to check if there's any difference (b'u')b

**Comment by Danial79** [comment]
I don't know how to load the new version, but I just tested the default cube and I'm getting about 30fps (even after a 2nd running). On my sculpt, however, it opened with 0.9, but dropped down to 0.36 as soon as I added a polystrip. On second loading, it was still at 0.36.

**Comment by carter2422** [comment]
@Danial79 were you able to test the speed after installing 2.0 with Blender set to Factory Defaults? This was how I initially tracked down the speed issues on my own laptop.

- Open Blender
- Load factory settings
- Install retopoflow
- Test speed

**Comment by QuietRezo** [comment]
@carter2422 I know that wasn't directly for me, but I tried those steps with no improvement. 

<img width="1917" alt="screen shot 2018-04-11 at 7 03 25 pm" src="https://user-images.githubusercontent.com/38005618/38649351-af96390c-3dbb-11e8-9098-c32ea3a888bd.png">

Do you guys have access to a computer running OSX for testing?

**Comment by carter2422** [comment]
Both of my machines are running OS X...

**Comment by carter2422** [comment]
Any chance you can share your model?

**Comment by vxlcoder** [comment]
Note: I also have access to an OSX machine, but I cannot reproduce this issue to any extent.  This is a very tricky problem, but I am working through it as quickly as I can.  I'll see if I can get access to some other OSX machines.

**Comment by QuietRezo** [comment]
Of course. Here's the .blend:
www.transfernow.net/79hi91l4wq8b
It's not finished yet, as I was testing on WIPs. And although the polycount isn't really high, it's still a DynTopo mesh, so it's too big to upload directly on GitHub's comments =_=

**Comment by vxlcoder** [comment]
OK, all... I'm still working on this problem, but I do have a workaround.

I have finally been able to reproduce it on a colleague's machine, and I've narrowed the problem down to a few lines of code.  I'm still working out why this is an issue, but I have coded a temporary workaround in the latest commit of `release/2.0.0` branch: change `Symmetry View` to `None`.  This workaround disables the symmetry visualization on the source, but will still mirror the target geometry.

I will update this thread once I have a better solution for the problem.

Thank you for your patience on this tricky problem!

**Comment by vxlcoder** [comment]
Hi all!
This is still an issue that I'm working on, but I wanted to give an update.
This issue appears to be larger than RF (meaning, it seems to affect Blender to a degree, too).  In an attempt to detect low FPS and gather data on why it is happening, I've added a dialog that will pop up if the framerate drops below a threshould.  (see commit 09ce032)
We will be releasing a Beta 2 version very soon, and it will contain the new dialog.
If you see this box show up, please consider posting your system details so we can attempt to find a solution.
Thank you for your help and patience!

**Comment by carter2422** [comment]
@vxlcoder just adding my laptop specs here for the `helmet-sculpt.blend` that's in our test-models folder.

On startup I get the low FPS warning, due to the many DynTopo sculpted pieces.

**Details:**

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Darwin, 17.5.0, Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64, x86_64, i386
- GPU: Intel Inc., Intel(R) Iris(TM) Plus Graphics 640, 2.1 INTEL-10.32.48, 1.20
- Target: verts:0, edges:0, faces:0
- Sources: number:19, verts:191523, edges:576254, faces:384861
- FPS: current:0.214845, threshold:10, time:2

**Comment by QuietRezo** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1060/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:62, edges:91, faces:30
- Sources: number:10, verts:184716, edges:555109, faces:369861
- FPS: current:0.670949, threshold:10, time:2

This isn't the same MacBook where I had issues initially, but after updating RF, I got the pop-up in this computer as well. After starting and orbiting around, FPS stabilized on 30fps. 
IIRC, it used to go up to 60, but 30 is fine and completely usable for retopology (b'u')b

**Comment by AssafNegev** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 62 Stepping 4, GenuineIntel
- GPU: ATI Technologies Inc., Radeon RX Vega, 4.5.13521 Compatibility Profile Context 24.20.11021.1000, 4.50
- Target: verts:0, edges:0, faces:0
- Sources: number:2, verts:324478, edges:973416, faces:648944
- FPS: current:1.368148, threshold:10, time:2

**Comment by vxlcoder** [comment]
Thank you for the reports!  Your specifications are very helpful!

Digging a little more into the context, though... does the frame rate dip only at times (like what you said @luicho06, that it happened at the start) but then pick back up again?  Or is RF entirely unusable due to its consistent slowness?

**Comment by kudamuchi** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.5 master 2018-07-07
- Platform: Windows, 10, 10.0.16299, AMD64, Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
- GPU: Intel, Intel(R) HD Graphics 4000, 4.0.0 - Build 10.18.10.4425, 4.00 - Build 10.18.10.4425
- Target: verts:6128, edges:12242, faces:6114
- Sources: number:7, verts:57744, edges:172115, faces:114388
- FPS: current:0.214159, threshold:10, time:2

**Comment by coyohti** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:411928, edges:1235774, faces:823849
- Sources: number:1, verts:1028, edges:3072, faces:2048
- FPS: current:0.124873, threshold:10, time:2

**Comment by billkakou** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:1, verts:426115, edges:1275492, faces:849441
- FPS: current:8.703930, threshold:10, time:2

**Comment by OscarLeif** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 8.1, 6.3.9600, AMD64, Intel64 Family 6 Model 63 Stepping 2, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 750/PCIe/SSE2, 4.6.0 NVIDIA 397.64, 4.60 NVIDIA
- Target: verts:222971, edges:668799, faces:445866
- Sources: number:4, verts:3288, edges:9722, faces:6448
- FPS: current:0.020898, threshold:1, time:5
Super slow.

**Comment by damian289** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD
- GPU: ATI Technologies Inc., AMD Radeon (TM) Pro WX 7100 Graphics, 4.5.13521 Compatibility Profile Context FireGL 24.20.11001.11006, 4.50
- Target: verts:1268, edges:2384, faces:1113
- Sources: number:2, verts:109833, edges:329748, faces:219802
- FPS: current:2.447415, threshold:10, time:2

**Comment by Sjockey** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 7, 6.1.7601, AMD64, Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 660/PCIe/SSE2, 4.6.0 NVIDIA 391.01, 4.60 NVIDIA
- Target: verts:1810, edges:3465, faces:1658
- Sources: number:1, verts:507410, edges:1522219, faces:1014811
- FPS: current:7.842690, threshold:10, time:2

**Comment by bkorkmaz** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:1, verts:589824, edges:1769472, faces:1179648
- FPS: current:1.950782, threshold:10, time:2

**Comment by vxlcoder** [comment]
Thank you everyone for your data!
Can you please edit your comments with a note stating about when RF runs slowly?  For example:

- RF runs slow all of the time (even with source meshes of < 10k triangles, for example)
- RF runs slow only at start up, but then runs fine
- RF is slow at "random" times
- etc.

Thank you for your help in figuring this issue out!

**Comment by galledark** [comment]
The issue: When RF runs slow all of the time (it starts when i activate any retopoflow tool)
My model is around 800 K Polys
I have the Low fps warning and a "Blender doesn't answer" issue every 5 seconds !


Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- GPU: ATI Technologies Inc., Radeon (TM) RX 480 Graphics, 4.5.13506 Compatibility Profile Context 23.20.15007.1005, 4.50
- Target: verts:332697, edges:998085, faces:665390
- Sources: number:1, verts:63492, edges:190464, faces:126976
- FPS: current:0.058079, threshold:10, time:2

**Comment by Mehd6384** [comment]
Running slow and submitting config as asked. 

Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 Unknown Unknown
- Platform: Linux, 4.15.0-29-generic, #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018, x86_64, x86_64
- GPU: NVIDIA Corporation, Quadro M2200/PCIe/SSE2, 4.6.0 NVIDIA 390.48, 4.60 NVIDIA
- Target: verts:210301, edges:630883, faces:420590
- Sources: number:1, verts:15931, edges:47787, faces:31858
- FPS: current:0.098813, threshold:10, time:2

**Comment by mdjustin** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:339, edges:617, faces:276
- Sources: number:1, verts:574958, edges:1724868, faces:1149912
- FPS: current:2.131075, threshold:10, time:2

**Comment by Puncia** [comment]
Mainly running slow when performing any action (it hangs up)

Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:777606, edges:2332824, faces:1555216
- Sources: number:1, verts:1006, edges:2972, faces:1968
- FPS: current:0.109535, threshold:10, time:2

**Comment by Danial79** [comment]
Quick question: How do I update to the latest build so I can get the pop-up? When I click the "Check now for retopoflow update" button, it comes up with "URL error, check internet connection".

**Comment by vxlcoder** [comment]
First off, what "error message" are you referring to, @Danial79?  Is it the Low FPS Warning or something else?

As a workaround, try setting the symmetry visualization to None _before_ turning on symmetry.  With the visualization off, you will still have symmetry, and you will still see the target mirrored, but the source will not be colored to show the symmetry.

![image](https://user-images.githubusercontent.com/3790699/43358329-d3663ec6-925d-11e8-9096-3569c5f073e1.png)

![image](https://user-images.githubusercontent.com/3790699/43358339-0a4a0080-925e-11e8-942b-ea641632205c.png)

We have made some recent changes that improve the overall performance here.  You can try installing the latest version of `release/2.0.0` branch, or wait just a few more days for us to officially release 2.0 on the Blender Market.

**Comment by vxlcoder** [comment]
Thank you everyone who has submitted their system details!

We are working hard at fixing this issue.  The latest version of `release/2.0.0` branch has some significant improvements, but I'm sure there are some areas that still need improvement.

If you know how to pull down a copy of RF from GitHub, give the `release/2.0.0` branch a try.  Otherwise, we plan to release officially RF 2.0 on Blender Market.

**Comment by Danial79** [comment]
@vlxcoder I was meaning the Low FPS pop-up. Sorry about that.

**Comment by SenseOfLife42** [comment]
Details:

- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080/PCIe/SSE2, 4.6.0 NVIDIA 398.11, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:1, verts:274530, edges:823584, faces:549056
- FPS: current:2.190135, threshold:10, time:2

**Comment by oORiggsOo** [comment]
low fps warning

**Comment by oORiggsOo** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Linux, 4.15.0-29-generic, #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018, x86_64, x86_64
- GPU: NVIDIA Corporation, GeForce GTX 960/PCIe/SSE2, 4.6.0 NVIDIA 390.48, 4.60 NVIDIA
- Target: verts:482, edges:992, faces:512
- Sources: number:1, verts:482, edges:1440, faces:960
- FPS: current:0.101772, threshold:5, time:10

**Comment by laaameche** [comment]
Low FPS
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080 Ti/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:63264, edges:189786, faces:126524
- Sources: number:6, verts:62343, edges:186831, faces:124542
- FPS: current:0.206885, threshold:5, time:10

**Comment by TeoLions** [comment]
It said that I should send report over low FPS, so that's what I'm doin'
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.6 experimental-build 2018-07-23
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 970/PCIe/SSE2, 4.6.0 NVIDIA 398.82, 4.60 NVIDIA
- Target: verts:253, edges:440, faces:190
- Sources: number:1, verts:6146, edges:18432, faces:12288
- FPS: current:2.284776, threshold:5, time:10

**Comment by Shakka57** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Darwin, 17.7.0, Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64, x86_64, i386
- GPU: ATI Technologies Inc., AMD Radeon R9 M370X OpenGL Engine, 2.1 ATI-1.68.20, 1.20
- Target: verts:0, edges:0, faces:0
- Sources: number:4, verts:835324, edges:2505796, faces:1670488
- FPS: current:2.735589, threshold:5, time:10

**Comment by LUK3D** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.6 blender2.8 2018-07-25
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GT 745M/PCIe/SSE2, 4.6.0 NVIDIA 390.77, 4.60 NVIDIA
- Target: verts:160, edges:244, faces:88
- Sources: number:1, verts:192267, edges:577762, faces:385476
- FPS: current:4.980036, threshold:5, time:10

**Comment by WadeWT** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:8074, edges:24189, faces:16117
- Sources: number:1, verts:176, edges:448, faces:272
- FPS: current:0.084401, threshold:5, time:10

**Comment by Coverop** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 398.36, 4.60 NVIDIA
- Target: verts:84789, edges:240738, faces:155951
- Sources: number:1, verts:8, edges:18, faces:12
- FPS: current:0.121723, threshold:5, time:10

**Comment by vxlcoder** [comment]
Thank you all for your info!

One thing I have noticed is that the target face count was very large (10k+) on some of your reports (@coyohti, @OscarLeif, @galledark, @Mehd6384, @Puncia, @laaameche, @WadeWT, @Coverop).  Did you accidentally start RF with the incorrect mesh selected and active (making it the target), or is this reporting the correct target mesh poly count?

RF is designed to create a low-poly "copy" of a high-poly mesh, and usually the low-poly version has a low poly count.

If it was an accident, I'll work in better detecting and reporting mechanisms to prevent this from happening.

As for the others (@QuietRezo, @luicho06, @AssafNegev, @kudamuchi, @billkakou, @EdmundDupont, @damian289, @Sjockey, @bkorkmaz, @mdjustin, @Danial79, @SenseOfLife42, @oORiggsOo, @TeoLions, @Shakka57, @LUK3D), especially those with fewer than 100 poly targets, did the performance every improve?  Or, did it continue to run slowly?

**Comment by mdjustin** [comment]
Hi @vxlcoder 
I seem to be getting fewer slow FPS warnings, Still getting some because I'm working with polygons near the millions, (Zbrush sculpts). So it seems to be better.

**Comment by TeoLions** [comment]
@vxlcoder yeah, I won't notice this lag at all, to be honest, if RF wouldn't bring report form.

**Comment by vxlcoder** [comment]
Two possible workarounds to improve performance is to add the decimate modifier and/or the remesh modifier your source meshes (no need to apply!).  These modifiers will simplify the source mesh, reducing the poly count, without changing the silhouette much.  You will lose some finer details, but a low-poly target won't capture these details without using a normal map or displacement map anyways.

Just a note, you can turn on/off the check in the options.  Simply check/uncheck `Perform Check` under Advanced > FPS Options

![image](https://user-images.githubusercontent.com/3790699/43791494-d46fb2f0-9a43-11e8-9996-7ea1278a4b30.png)

**Comment by coyohti** [comment]
Hi! There's only one mesh in my scene (well, two, but the other is a simple
UV sphere) and yes, it was really that high due to it being a full figure
(not broken down into parts). I was able to "solve" the issue using Sculpt
Toolkit to mask decimate regions that didn't need as much density but had
increased due to nearby higher density areas.

As a note, the areas that are dense on my object are just dense enough to
create the detail I need (example: eyelid crease) but not at a fine level
detail, like for wrinkles (which I would do on the low poly with multi-res
anyway). I was able to decimate these areas a little but not a whole lot
before they started losing detail in areas that I would be able to capture
in the low poly.

**Comment by laaameche** [comment]
@vxlcoder I made a mistake, my bad.

**Comment by Gr0m3kT** [comment]
Here comes Another one
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 7, 6.1.7601, AMD64, AMD64 Family 21 Model 19 Stepping 1, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 750 Ti/PCIe/SSE2, 4.6.0 NVIDIA 388.13, 4.60 NVIDIA
- Target: verts:8394, edges:19724, faces:11320
- Sources: number:16, verts:272046, edges:815948, faces:543936
- FPS: current:0.087845, threshold:5, time:10

**Comment by vxlcoder** [comment]
Some drilling questions to all:

- Do you have ambient occlusion turned on?
- Depth of Field?
- Matcaps?

![image](https://user-images.githubusercontent.com/3790699/43958018-ee07ee86-9c77-11e8-949c-8370eb97a84f.png)

If so, try using RF after turning them off.  Please report back whether changing settings helped or did not help, or that none of these render settings were enabled.

Thanks!

**Comment by DudeLazy** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.6 blender2.8 2018-08-06
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1050 Ti/PCIe/SSE2, 4.6.0 NVIDIA 388.75, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:2, verts:597555, edges:1793004, faces:1195461
- FPS: current:2.264395, threshold:5, time:10

**Comment by Julperado** [comment]
Hi, I just found this issue while testing retopoflow  2.0:
Details:
- RetopoFlow: 2.0.0
- Blender: 2.79.6 blender2.8 2018-08-20
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 398.82, 4.60 NVIDIA
- Target: verts:318, edges:577, faces:259
- Sources: number:3, verts:1201162, edges:3602578, faces:2401420
- FPS: current:1.028190, threshold:5, time:10
![imagen](https://user-images.githubusercontent.com/42665879/44567201-f973f980-a736-11e8-8478-768508d536ea.png)

**Comment by bartart3d** [comment]
Hello,
I just got this too, here are the specs:

Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 Unknown Unknown
- Platform: Linux, 4.15.0-33-lowlatency, #36-Ubuntu SMP PREEMPT Wed Aug 15 17:20:28 UTC 2018, x86_64, x86_64
- GPU: NVIDIA Corporation, Quadro 600/PCIe/SSE2, 4.6.0 NVIDIA 390.48, 4.60 NVIDIA
- Target: verts:7345, edges:14522, faces:7174
- Sources: number:2, verts:43964, edges:129569, faces:85587
- FPS: current:0.528298, threshold:5, time:10

**Comment by yvesbodson** [comment]
Details:
Well I think that the model has a too high poly count ..  I am going to have a lower count 
- RetopoFlow: 2.0.0 beta 2
- Blender: 2.79.0 master 2018-03-22
- Platform: Linux, 4.15.0-32-generic, #35-Ubuntu SMP Fri Aug 10 17:58:07 UTC 2018, x86_64, x86_64
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 396.51, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:2, verts:266911, edges:798701, faces:531778
- FPS: current:0.921328, threshold:10, time:2

**Comment by Maciek83** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 21 Model 2 Stepping 0, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 970/PCIe/SSE2, 4.6.0 NVIDIA 397.93, 4.60 NVIDIA
- Target: verts:512, edges:1003, faces:493
- Sources: number:1, verts:1439198, edges:4317576, faces:2878384
- FPS: current:1.472455, threshold:5, time:10

**Comment by vxlcoder** [comment]
To all that have posted here, can you please check if you have ambient occlusion or other post processing effects turned on? (See my previous comment for details)

**Comment by DudeLazy** [comment]
I did not have any of these enabled, also when I first opened RetopoFlow 2.0 it lagged for only the first five minutes after that it is working smooth even on the same high poly mesh.

These were the details I posted.

Details:

RetopoFlow: 2.0.0
Blender: 2.79.6 blender2.8 2018-08-06
Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
GPU: NVIDIA Corporation, GeForce GTX 1050 Ti/PCIe/SSE2, 4.6.0 NVIDIA 388.75, 4.60 NVIDIA
Target: verts:0, edges:0, faces:0
Sources: number:2, verts:597555, edges:1793004, faces:1195461
FPS: current:2.264395, threshold:5, time:10

**Comment by vxlcoder** [comment]
Thanks for info. Five minutes? Strange. I have a few other ideas on debugging this, but will wait for details from other posters.

**Comment by Julperado** [comment]
Hi Jon, yes in my case I had ambient occlusion enabled.
I don't know if this is relevant or not but the error appeared after roughly an hour of use, before that it was working pretty good, just felt slow when going in or out of "retopoflow mode".

**Comment by vxlcoder** [comment]
Thanks, @Julperado. We're you using symmetry and did you have symmetry visualization turned on?

**Comment by Julperado** [comment]
Yes, symmetry enabled and visualization turned on.

**Comment by Sarunas86** [comment]
Hi,
I wanted to test new retopoflow version. So just created a quick test object with ~2.000 000 Verts and enabled Polystrips tool. First I got message that mesh is to high poly and it took around 7 sec to activate the tool (but for a sclupt it is not that high poly). 
 **- what density of a mesh Retopoflow considers high poly?**
After the tool got active. I have made single polystrip and then activated mirror on X axis. then everything started to lag very badly..
My pc details are below
Very nice tool to have, just sad that it is still not working properly 

Details:
- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1080 Ti/PCIe/SSE2, 4.6.0 NVIDIA 399.07, 4.60 NVIDIA
- Target: verts:16, edges:22, faces:7
- Sources: number:1, verts:1821466, edges:5464408, faces:3642944
- FPS: current:4.072844, threshold:5, time:10

**Comment by vxlcoder** [comment]
Thank you for the report, @Sarunas86.  This is an issue that we are actively working on, but it is a tricky issue, because I haven't yet been able to reproduce it on any machine I have access to.

The lag _should_ subside shortly after enabling the symmetry (RF is collecting a bunch of data behind the scenes, which is causing the sluggishness).  Does RF run OK if you let it run for a while?  Also, you can skip the lag following enabling symmetry by first turning off the symmetry visualization (**None**, Edge, Face).

**Comment by brandiqa** [comment]
I got a low FPS warning. I guess am supposed to paste my specs here. Was doing something else then when I returned to Blender I got tthe message:
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 71 Stepping 1, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 960M/PCIe/SSE2, 4.6.0 NVIDIA 399.07, 4.60 NVIDIA
- Target: verts:0, edges:0, faces:0
- Sources: number:1, verts:472, edges:1416, faces:944
- FPS: current:3.734692, threshold:5, time:10

**Comment by vxlcoder** [comment]
@Visheshl, I'm sorry for your frustrations.  As you can tell by when this thread was first opened, we have been working on this problem for a while.  From what I can tell, this issue only affects a small number of machines.  And since none of the machines that I have been able to test RF on have shown this issue, I have not yet been able to find a cause.  I am currently on a retreat with CG Cookie, and some of the members have brought their laptops.  I will see if I can do some debugging while I'm here.

If you could, can you please post your computer's specifications, like CPU, memory, graphics card, version of OSX, etc.  Hopefully that will help me narrow down my search to the machines that are affected.

Thank you for your patience on this!  We are trying to resolve this issue as quickly as possible.

**Comment by SigismundSchlomoFreud** [comment]
Hi, all!
Help me, please, somebody!
I downloaded free RetopoFlow v 2.0 and I started work in it, but for some reason I can't move and rotate my viewport. Can anyone come across this?
I hope that somebody help me. Please!
P.S. Sorry for my English.

**Comment by SigismundSchlomoFreud** [comment]
Hello again!
Guess what?
.
.
.
My, gosh! I did it!
I just changed my native language to default in System.
Thank you very much for this wonderful program!
P.S. Maybe my problem will be useful to someone, so let these my comments remain here.

**Comment by vxlcoder** [comment]
@TJYSunset and @Creatortag21, does the slowness go away after a while, or is it consistently slow?

**Comment by TJYSunset** [comment]
It only freezes for seconds upon entering RF for the first time (and never occurs again until the addon is uninstalled in preferences), showing that dialog, regardless of viewport matcap & AO.

**Comment by Slir** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-02-21
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 416.94, 4.60 NVIDIA
- Target: verts:100, edges:183, faces:84
- Sources: number:18, verts:287939, edges:862929, faces:575105
- FPS: current:3.961084, threshold:5, time:10

**Comment by QbieShay** [comment]
Details:

- RetopoFlow: 2.0.1
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 21 Model 1 Stepping 2, AuthenticAMD
- GPU: ATI Technologies Inc., AMD Radeon R7 200 Series, 4.5.13416 Compatibility Profile Context 15.300.1025.1001, 4.40
- Target: verts:27829, edges:83475, faces:55648
- Sources: number:1, verts:964, edges:2880, faces:1920
- FPS: current:1.147270, threshold:5, time:10

**Comment by Lewiatan45** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080/PCIe/SSE2, 4.6.0 NVIDIA 399.07, 4.60 NVIDIA
- Target: verts:180, edges:340, faces:160
- Sources: number:1, verts:205647, edges:614393, faces:408760
- FPS: current:1.856076, threshold:5, time:10
- 48 GB RAM DDR4 3000 Mhz CL 15
- I7 6700k 4.4 Ghz

The Retopoflow 2.0.2 freezes Blender regularly. I retopo the model and suddenly Blender freezes and then slowly starts to respond with very choppy framerate and it goes back to normal. I have my Task Manager turned on and I don't see 100% CPU usage when the freeze occurs.

**Comment by AndiRutz** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1070 Ti/PCIe/SSE2, 4.6.0 NVIDIA 416.81, 4.60 NVIDIA
- Target: verts:128307, edges:384915, faces:256610
- Sources: number:3, verts:1446, edges:4320, faces:2880
- FPS: current:3.111537, threshold:5, time:10

**Comment by keianhzo** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 417.35, 4.60 NVIDIA
- Target: verts:24, edges:36, faces:12
- Sources: number:1, verts:776578, edges:2329728, faces:1553152
- FPS: current:3.887568, threshold:5, time:10

**Comment by Nit-Pick** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1070/PCIe/SSE2, 4.6.0 NVIDIA 416.34, 4.60 NVIDIA
- Target: verts:38068, edges:114198, faces:76132
- Sources: number:1, verts:482, edges:1440, faces:960
- FPS: current:1.041490, threshold:5, time:10

Modeling on a sculpted object

**Comment by Kalugin-Roman** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Linux, 4.15.0-43-generic, #46-Ubuntu SMP Thu Dec 6 14:45:28 UTC 2018, x86_64, x86_64
- GPU: NVIDIA Corporation, GeForce GTX 660/PCIe/SSE2, 4.6.0 NVIDIA 415.25, 4.60 NVIDIA
- Target: verts:44815, edges:89751, faces:44936
- Sources: number:3, verts:30467, edges:90657, faces:61439
- FPS: current:0.041402, threshold:5, time:10

**Comment by volkanayd** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1060 6GB/PCIe/SSE2, 4.6.0 NVIDIA 417.35, 4.60 NVIDIA
- Target: verts:730199, edges:2190271, faces:1460072
- Sources: number:1, verts:5284, edges:15840, faces:10560
- FPS: current:0.094719, threshold:5, time:10

**Comment by Hajruz** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1060/PCIe/SSE2, 4.6.0 NVIDIA 416.34, 4.60 NVIDIA
- Target: verts:1286, edges:2318, faces:1024
- Sources: number:6, verts:31181, edges:93262, faces:62084
- FPS: current:0.989951, threshold:5, time:10

**Comment by JohnRudy** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080/PCIe/SSE2, 4.6.0 NVIDIA 417.35, 4.60 NVIDIA
- Target: verts:61846, edges:185574, faces:123732
- Sources: number:3, verts:67500, edges:202130, faces:134642
- FPS: current:4.746973, threshold:5, time:10

**Comment by vxlcoder** [comment]
Hi all!  Thanks for the reports.  This has been a very elusive performance bug that I think I finally was able to reproduce (although I'm not entirely sure yet what I did to make that happen).

I want to try an experiment: After starting RF, press CTRL+UP to maximize the 3D view area.  Then report back here whether that improved the overall performance of RF or not.

Thanks for your help on this!

**Comment by vxlcoder** [comment]
@JohnRudy, @volkanayd, @Nit-Pick, @AndiRutz, @QbieShay:
Your target (low-res retopo mesh) model is really large.  Make sure that your active+selected object is the low-res version you want to continue retopo'ing.  All other visible meshes are considered source.

@Lewiatan45:
That might be happening when Blender auto-saves the file.  Try adjusting your auto-save timing in the Blender preferences.  However, let me know if the pref setting isn't set to 120seconds / 2minutes now, and I'll take a deeper look into the problem.

**Comment by kenpoh01** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17763, AMD64, Intel64 Family 6 Model 30 Stepping 5, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1050 Ti/PCIe/SSE2, 4.6.0 NVIDIA 417.71, 4.60 NVIDIA
- Target: verts:5821, edges:11416, faces:5597
- Sources: number:1, verts:28889, edges:85999, faces:57109
- FPS: current:2.816907, threshold:5, time:10

**Comment by Shill1985** [comment]
@vxlcoder 
maximizing 3d view area made a big difference for me. 10 fps before, 30+ after

**Comment by vxlcoder** [comment]
That's great to read, @Shill1985!  Working on a better solution, but I hope that this is not an unhelpful workaround in the meantime.

**Comment by vxlcoder** [comment]
All, I think that commit d824347 in the `release/2.0.3` branch will fix most of these problems!

Please test this if you get a chance, and let me know whether it works for you or not.

We'll be releasing v2.0.3 officially soon.

**Comment by ProfessorWoland** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.7 blender2.7 2019-02-25
- Platform: Linux, 4.15.0-45-generic, #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019, x86_64, x86_64
- GPU: Intel Open Source Technology Center, Mesa DRI Intel(R) Sandybridge Mobile , 3.0 Mesa 18.2.2, 1.30
- Target: verts:16, edges:16, faces:0
- Sources: number:1, verts:482, edges:1440, faces:960
- FPS: current:1.961483, threshold:5, time:10

I can't use it at all! Does I have to own discrete GPU integrated isn't enough or?

**Comment by vxlcoder** [comment]
Thanks for the report, @ProfessorWoland.  Can you try out the `release/2.0.3` branch to see if it improves your performance?  I'm looking for testers now

**Comment by vxlcoder** [comment]
All:

Below is a link to a preview of version 2.0.3.  We are working on officially releasing it now, but getting feedback and testers on it would be helpful.  Thanks!

https://www.dropbox.com/s/opc58w0s5fmxbou/RetopoFlow_v2.0.3.zip?dl=0

**Comment by ProfessorWoland** [comment]
Just tried 2.0.3 and it makes no difference what so ever, as soon as I make a stroke with any 'brush' it freezes, and pops up a message to report a problem, what I Have noticed is if I close that pop-up is that blender is using 100% of CPU and I can't even access a system monitor to kill it, only way is to hold down power button on pc to shut down, however if I don't close the pop up I blender doesn't hogs the CPU and I can use pc normally as long as pop-up is there, this goes for both versions, keep in mind that my CPU is i5 2520m so 2c, 4t, it is low-end and integrated GPU hd3000 so I guess you need to have some decent setup in order to use addon. If you need any more info, speak.

**Comment by vxlcoder** [comment]
Sorry to hear this, @ProfessorWoland!

From what I've gathered online, your machine seems to be from 2011 or so.  You could be right in that your machine's specs could be the cause of what you're seeing (I haven't tested RF on a machine with those specs, so I'm not completely sure, yet). Does Blender run fine with a high-res model in edit mode?  For example, can you load and edit a model with 10k or 100k verts without much problems?

RF does a lot of drawing (UI, symmetry, brushes, etc.), and I'll admit that the drawing code isn't the best it could be (working on that now), but I wouldn't think that it would cause such an issue.  However, that might be the cause.

My hunch though is that the issue is in the way we're doing background processing.  Try this to see if it helps: start RF, then Options > Advanced > uncheck Async Loading.  Then, close and restart Blender, and see if RF allows you to retopo.

**Comment by ProfessorWoland** [comment]
Nope, this didn't help it still the same. Yeah you can call hardware ancient, however I didn't have issues using blender if I keep it under some number. In edit mode I've never tried editing mesh with 10k vertices (I did now with subdivided cube ~24k vertices and I didn't have issues, I tried proportional editing and didn't have issues) however when I enter edit mode with sculpt (dynotopo) with around 350k verts I have hickups when orbiting around and working in such conditions isn't pleasant or even possible, proportional editing starts reacting after 10-15 seconds. 
I Think my problem in case of RF is GPU if I had decent GPU it probably won't have that much problems or at all, but combination of not so perfect code and my hardware is, well..
In other hand I see some of people who complained have 1080 cards and have the problem too, so I'm not sure what is it, in my case it's hardware 90%.

**Comment by vxlcoder** [comment]
Note to all: we've started a new GitHub issue (#710) for this performance bug, because this thread has become long and difficult to determine what is the current state.  If you're still seeing the low FPS warning, be sure to update to the latest official version of RetopoFlow (2.0.3) to see if that fixes the issue.  Otherwise, post your details to the new GH issue (#710).  Thank you all for your feedback and patience so far.  We're trying to fix this bug completely, but it is a very tricky bug to find!

**Comment by OpticDreams** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 1080 Ti/PCIe/SSE2, 4.6.0 NVIDIA 419.35, 4.60 NVIDIA
- Target: verts:1021, edges:1920, faces:897
- Sources: number:2, verts:10261, edges:30273, faces:20067
- FPS: current:2.396935, threshold:5, time:10

**Comment by vxlcoder** [comment]
Thanks for the post @OpticDreams. Please update to RF v2.0.3. We fixed many things, and this performance bug is one we worked on. Thanks!

**Comment by ArtistArtorias** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 970/PCIe/SSE2, 4.6.0 NVIDIA 416.16, 4.60 NVIDIA
- Target: verts:18, edges:27, faces:10
- Sources: number:1, verts:1582864, edges:4746276, faces:3163502
- FPS: current:2.860978, threshold:5, time:10

**Comment by vxlcoder** [comment]
Thanks for the post @ArtistArtorias. Please update to RF v2.0.3. We fixed many things, and this performance bug is one we worked on. Thanks!

**Comment by OpticDreams** [comment]
> Thanks for the post @OpticDreams. Please update to RF v2.0.3. We fixed many things, and this performance bug is one we worked on. Thanks!

The update fixed the problem. Thanks. :)

**Comment by Syndaryl** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2017-09-11
- Platform: Windows, 10, 10.0.17763, AMD64, AMD64 Family 23 Model 1 Stepping 1, AuthenticAMD
- GPU: NVIDIA Corporation, GeForce GTX 1080 Ti/PCIe/SSE2, 4.6.0 NVIDIA 417.35, 4.60 NVIDIA
- Target: verts:13791, edges:27585, faces:13796
- Sources: number:13, verts:4066575, edges:12199025, faces:8132483
- FPS: current:0.043223, threshold:5, time:10


I'm updating to 2.0.3 now, thank you for this issue :)

**Comment by vxlcoder** [comment]
Thanks for the report, @Syndaryl.  Please let us know if 2.0.3 fixes the performance issue.

Another thing to try is to decimate your sources.  8m faces is pretty high, and you probably don't need all that resolution just to retopo.  RF is tested to run will up to about 2m faces (depending on the machine), so if you can reduce the poly count with the decimate modifier, you might see a huge performance boost.

Either way, let us know what fixes your problems or if your problems still exist.  Thanks!

**Comment by byw** [comment]
Details:

- RetopoFlow: 2.0.2
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17763, AMD64, Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
- GPU: NVIDIA Corporation, GeForce GTX 680MX/PCIe/SSE2, 4.6.0 NVIDIA 391.01, 4.60 NVIDIA
- Target: verts:184, edges:289, faces:109
- Sources: number:1, verts:117531, edges:352587, faces:235058
- FPS: current:2.100151, threshold:5, time:10

**Comment by 00buggy00** [comment]
Details:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Darwin, 17.7.0, Darwin Kernel Version 17.7.0: Tue Feb 18 22:51:29 PST 2020; root:xnu-4570.71.73~1/RELEASE_X86_64, x86_64, i386
- GPU: Intel Inc., Intel HD Graphics 3000 OpenGL Engine, 2.1 INTEL-10.4.14, 1.20
- Target: verts:0, edges:0, faces:0
- Sources: number:19, verts:748485, edges:2244715, faces:1496362
- FPS: current:0.085555, threshold:5, time:10

**Comment by jlampel** [comment]
Closing this out as it should have been fixed in 2.0.3. If you have any performance related issues in version 3 or higher, please let us know and open a new issue. Thanks!
